src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -57,7 +56,7 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
 
     val blockchain = BlockchainImpl(storages)
     val parentStateRoot = blockchain.getBlockHeaderByHash(block.header.parentHash).map(_.stateRoot)
-    val initialWorld = InMemoryWorldStateProxy(storages, blockchainConfig.accountStartNonce, parentStateRoot)
+    val initialWorld = blockchain.getReadOnlyWorldStateProxy(-1, blockchainConfig.accountStartNonce, parentStateRoot)
```

Comment:
Yes! after adding `ReadOnlyNodeStorage` I'm able to change this to optional. Good catch!